src/main.rs

```diff
@@ -1,18 +1,9 @@
-extern crate clap;
-extern crate colored;
 #[macro_use]
 extern crate error_chain;
-extern crate ethabi;
-#[macro_use(EthabiContract)]
-extern crate ethabi_derive;
 #[macro_use(use_contract)]
 extern crate ethabi_contract;
-extern crate parse_duration;
-extern crate serde;
-#[macro_use(Deserialize)]
-extern crate serde_derive;
-extern crate serde_json;
-extern crate web3;
+#[macro_use(EthabiContract)]
```

Comment:
Import the macros with `use` and throw out the `extern`s?

Reply:
Hmm idk, but one `extern` is nothing to worry about. I tried building with `1.33.0-nightly` and `RUSTFLAGS="-Z external-macro-backtrace"` to debug the error and got this macro backtrace for each of the `use_contract!`s:
```
error: cannot find derive macro `EthabiContract` in this scope
  --> <::ethabi_contract::use_contract macros>:5:14
   |
1  | / ( $ module : ident , $ path : expr ) => {
2  | | # [ allow ( dead_code ) ] # [ allow ( missing_docs ) ] # [
3  | | allow ( unused_imports ) ] # [ allow ( unused_mut ) ] # [
4  | | allow ( unused_variables ) ] pub mod $ module {
5  | | # [ derive ( EthabiContract ) ] # [ ethabi_contract_options ( path = $ path )
   | |              ^^^^^^^^^^^^^^ help: try: `ethabi_contract`
6  | | ] struct _Dummy ; } }
   | |_____________________- in this expansion of `use_contract!`
   | 
  ::: src/contracts.rs:25:5
   |
25 |       use_contract!(voting, "abi/v1/VotingToChangeKeys.abi.json");
   |       ------------------------------------------------------------ in this macro invocation
```

Reply:
Yes, I think it's because `use_contract!` uses `derive(EthabiContract)` _inside_ a module, in which it doesn't `use EthabiContract`: https://github.com/paritytech/ethabi/blob/ad77c92ec7cf735d702f19915c3a4cd6c83a4cb4/contract/src/lib.rs#L11
If it did, it wouldn't be usable with Rust 2015 anymore.
I guess the promise to seamlessly mix Rust 2015 and 2018 crates can't quite be kept for all macros. :pensive: